apps/browser/src/autofill/background/overlay.background.ts

```diff
@@ -68,6 +100,109 @@ class OverlayBackground {
     await this.getAuthStatus();
   }
 
+  /**
+   * Updates the overlay list's ciphers and sends the updated list to the overlay list iframe.
+   * Queries all ciphers for the given url, and sorts them by last used. Will not update the
+   * list of ciphers if the extension is not unlocked.
+   */
+  async updateOverlayCiphers() {
+    if (this.userAuthStatus !== AuthenticationStatus.Unlocked) {
+      return;
+    }
+
+    const currentTab = await BrowserApi.getTabFromCurrentWindowId();
+    if (!currentTab?.url) {
+      return;
+    }
+
+    this.overlayLoginCiphers = new Map();
+    const ciphersViews = (await this.cipherService.getAllDecryptedForUrl(currentTab.url)).sort(
+      (a, b) => this.cipherService.sortCiphersByLastUsedThenName(a, b)
+    );
+    for (let cipherIndex = 0; cipherIndex < ciphersViews.length; cipherIndex++) {
+      this.overlayLoginCiphers.set(`overlay-cipher-${cipherIndex}`, ciphersViews[cipherIndex]);
+    }
+
+    const ciphers = this.getOverlayCipherData();
+    this.overlayListPort?.postMessage({ command: "updateOverlayListCiphers", ciphers });
+    await BrowserApi.tabSendMessageData(currentTab, "updateIsOverlayCiphersPopulated", {
+      isOverlayCiphersPopulated: Boolean(ciphers.length),
+    });
+  }
+
+  /**
+   * Strips out unnecessary data from the ciphers and returns an array of
+   * objects that contain the cipher data needed for the overlay list.
+   */
+  private getOverlayCipherData(): OverlayCipherData[] {
+    const isFaviconDisabled = this.settingsService.getDisableFavicon();
+    const overlayCiphersArray = Array.from(this.overlayLoginCiphers);
+    const overlayCipherData = [];
+    let loginCipherIcon: WebsiteIconData;
+
+    for (let cipherIndex = 0; cipherIndex < overlayCiphersArray.length; cipherIndex++) {
+      const [overlayCipherId, cipher] = overlayCiphersArray[cipherIndex];
+      if (!loginCipherIcon && cipher.type === CipherType.Login) {
+        loginCipherIcon = WebsiteIconService.buildCipherIconData(
+          this.iconsServerUrl,
+          cipher,
+          isFaviconDisabled
+        );
+      }
+
+      overlayCipherData.push({
+        id: overlayCipherId,
+        name: cipher.name,
+        type: cipher.type,
+        reprompt: cipher.reprompt,
+        favorite: cipher.favorite,
+        icon:
+          cipher.type === CipherType.Login
+            ? loginCipherIcon
+            : WebsiteIconService.buildCipherIconData(
+                this.iconsServerUrl,
+                cipher,
+                isFaviconDisabled
+              ),
+        login:
+          cipher.type === CipherType.Login
+            ? { username: this.getObscureName(cipher.login.username) }
+            : null,
+        card:
+          cipher.type === CipherType.Card
+            ? { brand: cipher.card.brand, partialNumber: `*${cipher.card.number?.slice(-4)}` }
```

Comment:
nit

```suggestion
            ? { brand: cipher.card.brand, partialNumber: cipher.card.number ? `*${cipher.card.number.slice(-4)}` : null }
```

or (like we do in extension and context menu):

```suggestion
            ? { brand: cipher.card.brand, partialNumber: cipher.card.subTitle }
```